packages/code-infra/src/cli/babel.mjs

```diff
@@ -0,0 +1,293 @@
+      pluginTransformInlineEnvVars,
+      {
+        include: [
+          'MUI_VERSION',
+          'MUI_MAJOR_VERSION',
+          'MUI_MINOR_VERSION',
+          'MUI_PATCH_VERSION',
+          'MUI_PRERELEASE',
+        ],
+      },
+    ],
+  ];
+
+  if (removePropTypes) {
+    plugins.push([
+      pluginRemovePropTypes,
+      {
+        mode: 'unsafe-wrap',
+      },
+    ]);
+  }
+
+  if (errorCodesPath) {
+    plugins.push([
+      pluginMinifyErrors,
+      {
+        missingError,
+        errorCodesPath,
+        runtimeModule: '@mui/utils/formatMuiErrorMessage',
+      },
+    ]);
+  }
+
+  if (optimizeClsx) {
+    plugins.push([pluginOptimizeClsx]);
+  }
+
+  if (searchAndReplaceModule) {
+    const pluginOptions = await import(searchAndReplaceModule);
+    plugins.push([pluginSearchAndReplace, pluginOptions.default]);
+  }
+
+  if (bundle === 'esm') {
+    plugins.push([pluginResolveImports, { outExtension }]);
+  }
+
+  return {
+    assumptions: {
+      noDocumentAll: true,
+      // With our case these assumptions are safe, and the
+      // resulting behavior is equivalent to spec mode.
+      setPublicClassFields: true,
+      privateFieldsAsProperties: true,
+      objectRestNoSymbols: true,
+      setSpreadProperties: true,
+    },
+    ignore: [
+      // Fix a Windows issue.
+      /@babel[\\|/]runtime/,
+      // Fix const foo = /{{(.+?)}}/gs; crashing.
+      /prettier/,
+      '**/*.template.js',
+    ],
+    presets: [
+      [presetEnv, presetEnvOptions],
+      [presetReact, { runtime: 'automatic' }],
+      [presetTypescript],
+    ],
+    plugins,
+  };
+}
+
+/**
+ * @param {Object} options
+ * @param {boolean} [options.verbose=false] - Whether to enable verbose logging.
+ * @param {boolean} [options.optimizeClsx=false] - Whether to enable clsx call optimization transform.
+ * @param {boolean} [options.removePropTypes=true] - Whether to enable removal of React prop types.
+ * @param {BuildConfig} [options.buildConfig] - The build configuration.
+ * @param {string[]} [options.ignores] - The globs to be ignored by Babel.
+ * @param {string} options.pkgVersion - The package version.
+ * @param {string} options.sourceDir - The source directory to build from.
+ * @param {string} options.outDir - The output directory for the build.
+ * @param {string} options.outExtension - The output file extension for the build.
+ * @param {boolean} options.hasLargeFiles - Whether the build includes large files.
+ * @param {BundleType} options.bundle - The bundles to build.
+ * @param {string} options.babelRuntimeVersion - The version of @babel/runtime to use.
+ * @returns {Promise<void>}
+ */
+export async function babelBuild({
+  sourceDir,
+  outDir,
+  babelRuntimeVersion,
+  hasLargeFiles,
+  bundle,
+  buildConfig,
+  pkgVersion,
+  outExtension,
+  optimizeClsx = false,
+  removePropTypes = false,
+  verbose = false,
+  ignores = [],
+}) {
+  const files = await globby(['**/*.{cjs,js,mjs,cts,ts,mts,cjsx,ctsx,tsx,mtsx}'], {
```

Comment:
looks like we have a few dependencies on `fast-glob` in the repo, but it seems unused.

Reply:
`globby` is a wrapper over `fast-glob`. We can use either but `code-infra` already had `globby` so I used this.